### ISSUE TITLE
Train the model with input shape of 256 x 512 rather than 288 x 800 

### ISSUE THREAD

**Issue by letdivedeep** [cfzd/Ultra-Fast-Lane-Detection#124]
I wanted to train the model with an input shape of 256 x 512 rather than 288 x 800. While changing the shape I get lots of errors with the unsupported shape. How do we try to make this transition easy and code to work for every different input shape?  or u recommend converting all FC layers to conv

**Comment by cfzd** [comment]
@letdivedeep 
Just change the dim of FC according to your resolution should work. For `288x800`, the size of the feature map is `8x(288/32=9)x(800/32=25) = 1800`. You can adjust the dim of FC using the same rule.

Moreover, you need to change the row anchors correspondingly. In `data/constant.py`, you should adjust the row anchors using `new_x = x / 288 * 256`, in which `x` is the elements in the row anchors.

**Comment by letdivedeep** [comment]
@cfzd  Thanks for your reply 

I also wanted to change the backbone to mobilenetv3 large or efficientNet B2 with reduced input dim. Quite confused about how to do split layers for x2,x3,x4 branch, to do it randomly and how will this affect the overall architecture

**Comment by cfzd** [comment]
@letdivedeep 
Sorry, I forget the segmentation branch. If you just want to test if it works, the fastest way is to set `use_aux = False` in the `config`. Then the segmentation branch is disabled and the size mismatch won't happen.

If you also want to add the segmentation branch, you can change here:
https://github.com/cfzd/Ultra-Fast-Lane-Detection/blob/bb673eb816e5c5356dcab02d72feb937fbb445b6/data/dataloader.py#L15
For example, `36=288/8` and `100=800/8`. Just change the size according to your input size should work.

**Comment by letdivedeep** [comment]
@cfzd Thanks for the help and awesome work. I did the changes as per your suggestion and it works 

one last thing i may need to ask 
> Little bit confused on the changes you suggested to make in data/constant.py

`Moreover, you need to change the row anchors correspondingly. In data/constant.py, you should adjust the row anchors using new_x = x / 288 * 256, in which x is the elements in the row anchors.`

Is the number of elements in row anchors, the same as the number of griding cells(eg:100) 
```
tusimple_row_anchor = [ 64,  68,  72,  76,  80,  84,  88,  92,  96, 100, 104, 108, 112,
            116, 120, 124, 128, 132, 136, 140, 144, 148, 152, 156, 160, 164,
            168, 172, 176, 180, 184, 188, 192, 196, 200, 204, 208, 212, 216,
            220, 224, 228, 232, 236, 240, 244, 248, 252, 256, 260, 264, 268,
            272, 276, 280, 284]
```
 if we are using dim of 256x 512 cant we just from 32, 36 .....252

Is my understanding correct ?

**Comment by cfzd** [comment]
@letdivedeep 
The number of row anchors has no relation to the number of griding cells. It just means how many points you want to represent a lane. 

The correct row anchor should be:
```
tusimple_row_anchor_h256 = [ 56,  60,  64,  67,  71,  74,  78,  81,  85,  88,  92,  96,  99,
       103, 106, 110, 113, 117, 120, 124, 128, 131, 135, 138, 142, 145,
       149, 152, 156, 160, 163, 167, 170, 174, 177, 181, 184, 188, 192,
       195, 199, 202, 206, 209, 213, 216, 220, 224, 227, 231, 234, 238,
       241, 245, 248, 252]
```

For example, the first element `56 = int(64 * 256 / 288)`. The last one `252 = int(284 * 256 / 288)`. Just use linear scale should work.

**Comment by letdivedeep** [comment]
@cfzd I really appreciate your help and support provided to clarify my doughts. Thanks a lot

**Comment by cfzd** [comment]
Sorry, the previous method is not exactly correct. I will make a refactor of the row anchor system to support different resolutions.

**Comment by emtsrlee** [comment]
> Sorry, the previous method is not exactly correct. I will make a refactor of the row anchor system to support different resolutions.

Waiting for your new release.
Thank you

**Comment by eriche2016** [comment]
@cfzd  Hi, nice work.  BTW, when will the version that supports row anchor system of different resolutions release?

**Comment by eriche2016** [comment]
I am new to this field,  and also I am curious about the reasons why the linear scale solutiondoes not work.

**Comment by AbdulFMS** [comment]
can you provide the pretrained model with mobilenetv2 and the new shape 256*512 please! @letdivedeep
